mockingbird-compiler/src/main/kotlin/com/careem/mockingbird/MockingbirdPlugin.kt

```diff
@@ -54,17 +55,18 @@ abstract class MockingbirdPlugin : Plugin<Project> {
                 EXTENSION_NAME, MockingbirdPluginExtensionImpl(target.objects)
             )
 
-            target.task(GradleTasks.GENERATE_MOCKS) {
-                dependsOn(target.tasks.getByName(GradleTasks.ASSEMBLE))
-                doLast {
-                    generateMocks(target)
-                }
-            }
-
             target.gradle.projectsEvaluated {
+                val generateMocksTask = target.task(GradleTasks.GENERATE_MOCKS) {
+                    dependsOn(target.tasks.getByName(GradleTasks.JVM_JAR))
+                    doLast {
+                        generateMocks(target)
+                    }
+                }
 
-                target.tasks.getByName(GradleTasks.ALL_TESTS) {
-                    dependsOn(target.tasks.getByName(GradleTasks.GENERATE_MOCKS))
+                target.tasks.forEach { task ->
+                    if (task.name.contains("Test") && (task is KotlinCompile)) {
```

Comment:
Some check is used internally to distinguish test and non-test compilations: https://github.com/JetBrains/kotlin/blob/d77041aad06a4a1810b641a20dd8162a14b5ba28/libraries/tools/kotlin-gradle-plugin/src/common/kotlin/org/jetbrains/kotlin/gradle/tasks/Tasks.kt#L350.